gridmet.py

```diff
@@ -0,0 +1,152 @@
+def g2shp_regridding(polygon_file_path,
+                     output_data_folder,
+                     var_short,
+                     start_date, end_date,
+                     lon_min, lon_max, lat_min, lat_max,
+                     g2s_file_prefix = 'tmp'):
+    """
+    :param str polygon_file_path: path to hru shapefile used for regridding of gridMET data
+    :param str output_data_folder: folder location where outputs such as weights pickle file and netcdf output file will reside
+    :param str/list var_short: data variable short name or list of data variable short names. Must be one of the following: ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    :param str start_date: Start date of data collection yyyy-mm-dd
+    :param str end_date: End date of data collection yyyy-mm-dd
+    :param str lon_min: bbox of aoi longitude min
+    :param str lon_max: bbox of aoi longitude max
+    :param str lat_min: bbox of aoi latitude min
+    :param str lat_max: bbox of aoi latitude max
+    :param str g2s_file_prefix: prefix used to save the grd2shp_xarray
+    :return: g2s array
+    """
+
+    ## Create dictionary of short variable names with official long variable name
+    data_vars_shrt_all = ['tmmx', 'tmmn', 'pr', 'srad', 'vs', 'rmax', 'rmin', 'sph']
+    data_vars_long_all = ['daily_maximum_temperature', 'daily_minimum_temperature',
+                          'precipitation_amount', 'daily_mean_shortwave_radiation_at_surface',
+                          'daily_mean_wind_speed', 'daily_maximum_relative_humidity',
+                          'daily_minimum_relative_humidity', 'daily_mean_specific_humidity']
+
+    all_data_vars_dict = dict(zip(data_vars_shrt_all, data_vars_long_all))
+    subset = {key: all_data_vars_dict[key] for key in var_short}
+    print(subset)
+    del all_data_vars_dict
+
+    ## geospatial vector file
+    gdf = gpd.read_file(polygon_file_path)
+
+    ## check if only 1 var (short and long) is provided, change to list
+    if type(var_short) == str:
+        var_short = [var_short]
+        print(var_short)
+    ## check that var_short is a list
+    elif type(var_short) != list:
+        var_short = [var_short]
+        print(var_short)
+
+    ## empty list for the xarray.datasets
+    data_xr_ds_list = []
+
+    ## Creating ds
+    for i in var_short:
+        # Get url:
+        print(i)
+        ## source: http://thredds.northwestknowledge.net:8080/thredds/reacch_climate_MET_aggregated_catalog.html
+        url = 'http://thredds.northwestknowledge.net:8080/thredds/dodsC/agg_met_{}_1979_CurrentYear_CONUS.nc'.format(i)
+        print(url)
+
+        # call data from url
+        start = time.perf_counter()
+        df = xr.open_dataset(url + '#fillmismatch')
+
+        # Subset to timeframe and bbox of interest:
+        df_subset = df.sel(day=slice(start_date, end_date),
+                           lon=slice(lon_min, lon_max),
+                           lat=slice(lat_max, lat_min))
```

Comment:
I suggest using `ds` (for dataset) instead of `df` here. That's the convention I've seen (`df` = pandas DataFrame; `ds` = xarray DataSet)
```suggestion
        ds = xr.open_dataset(url + '#fillmismatch')

        # Subset to timeframe and bbox of interest:
        ds_subset = ds.sel(day=slice(start_date, end_date),
                           lon=slice(lon_min, lon_max),
                           lat=slice(lat_max, lat_min))
```